src/lib/pbkdf/pbkdf.h

```diff
@@ -147,6 +161,20 @@ class BOTAN_DLL PBKDF
          }
    };
 
+/**
+* Password based key derivation function factory method
+* @param algo_spec the name of the desired PBKDF algorithm
+* @return pointer to newly allocated object of that type
+*/
+inline PBKDF* get_pbkdf(const std::string& algo_spec,
+                           const std::string& provider = "")
+   {
+   std::unique_ptr<PBKDF> p(PBKDF::create(algo_spec, provider));
+   if(p)
```

Comment:
In my opintion it is too easy to change a code like that

```
if (p)
    p->doOneThink();
```

to

```
if (p)
    p->doOneThink();
    p->doAnotherThink();
```

without any notice by the compiler or anything. That risk can be reduced dramatically by requiring

```
if (p)
   {
   p->doOneThink();
   }
```

or

```
if (p) p->doOneThink();
```

Especially the later versions fits great for null pointer checks like in that case.